presto-main-base/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -799,7 +799,8 @@ public PlanOptimizers(
                 ruleStats,
                 statsCalculator,
                 estimatedExchangesCostCalculator,
-                new GroupInnerJoinsByConnectorRuleSet(metadata, predicatePushDown).rules()));
+                new GroupInnerJoinsByConnectorRuleSet(metadata).rules()));
+        builder.add(predicatePushDown, simplifyRowExpressionOptimizer);
```

Comment:
Instead of pulling this out completely, we could try rewriting the left deep tree as one without any GroupJoin nodes. I will leave a comment in the `GroupInnerJoinsByConnectorRuleSet` with something that you could try